### ISSUE TITLE
Provide manylinux binary wheels for linux

### ISSUE THREAD

**Issue by TkTech** [TkTech/pysimdjson#12]
Trivial, but time consuming. simdjson uses c++17 features in a few places that make it depend on recent versions of gcc and libstdc++. Normally for 99% of C/C++ bindings we can get away with just using the manylinux docker images, but only when we can use gcc5. We need to build gcc7 on centos5 and static link libstc++ (which will add a couple MB to the binary 😢).

Building gcc is not what I'd call "fast", so we definitely need to create our own docker image with it ready to go. I imagine this would definitely be useful to other projects, so worth the effort.

@lemire tweet in https://twitter.com/lemire/status/1100566630994792448 makes it pretty clear interested parties aren't always going to know how to deal with a C compiler. Actually pretty happy about that, since it means Linux is becoming more accessible. Tilts the scale from "eh, linux users know the routine" to "we should probably do this."

Refs:
- https://github.com/pypa/manylinux/issues/118
- https://www.python.org/dev/peps/pep-0571/
- https://github.com/pypa/manylinux/issues/179
- https://www.python.org/dev/peps/pep-0491/

**Comment by TkTech** [comment]
tl;dr: x86_64 `manylinux` packages now available.

I've created a new docker image based off of the `manylinux1_x86_64.` image from quay. This image will grab and compile a GCC specified by `$GCC_VERSION`. It is strongly recommended to use the pre-built image I've put up on docker hub, as building gcc can take hours, and will turn a metal macbook pro into a flesh-melting fire risk. GCC is compiled with c, c++, and fortran (for numpy) support.

This image is capable of generating manylinux binaries that use recent gcc features such as C++14/C++17/some C++2a.

@lemire @geofflangdale please give the new packages a try.

**Comment by lemire** [comment]
Wow. This handles AVX2 issues at runtime, right?

(It works on my various machines but I don't think I have a non-x64 machines without AVX2.)
